File: planning-idsdp-rms-rpm-backend/src/main/java/com/ibcs/idsdp/rpm/model/domain/ResearcherProposalBudgetDetails.java
package com.ibcs.idsdp.rpm.model.domain;

import javax.persistence.*;
import javax.validation.constraints.NotNull;

import org.hibernate.annotations.Cache;
import org.hibernate.annotations.CacheConcurrencyStrategy;

import com.ibcs.idsdp.common.model.domain.BaseEntity;

import lombok.Data;

@Table(name = "m1_researcher_proposal_budget_details")
@Entity
@Data
@Cache(usage = CacheConcurrencyStrategy.READ_WRITE)
public class ResearcherProposalBudgetDetails extends BaseEntity{

	@NotNull
	@ManyToOne
	@JoinColumn(name = "researcher_proposal_id")
	private ResearcherProposal researcherProposal;

	@NotNull
	private Long stExpenditureItemId;

	private String purpose;

	@NotNull
	private Double totalAmount;

	private Integer isEditable;
	
	private String expenditureName;

}

File: planning-idsdp-rms-frontend/src/app/main/modules/rpm/models/ResearcherProposalBudgetDetails.ts
export class ResearcherProposalBudgetDetails {

    id: number;
    uuid: string;
    researcherProposalId: number;
    stExpenditureItemId: any;
    purpose: string;
    totalAmount: number;
    isEditable: 1 | 0;
    isDeleted: 1 | 0;
    expenditureName: string;
    expenditureItem?: any;
    showExpenditureItem: boolean;

}
